File: Microfulfilment.Web/ClientApp/src/app/models/product.ts
export class Product {
    public id: number;
    public name: string;
    public barcode: string;
    public retailPrice: number;
    public costPrice: number;
    public stock: number;
    public minimumQuantity: number;
    public maximumQuantity: number;

    constructor(args) {
        this.id = args.id;
        this.name = args.name;
        this.barcode = args.barcode;
        this.retailPrice = args.retailPrice;
        this.costPrice = args.costPrice;
        this.stock = args.stock;
        this.minimumQuantity = args.minimumQuantity;
        this.maximumQuantity = args.maximumQuantity;
    }
}

File: Microfulfilment.Models/Product.cs
namespace Microfulfilment.Models
{
    public class Product
    {
        public int Id { get; set; }

        public string Name { get; set; }

        public string Barcode { get; set; }

        public decimal RetailPrice { get; set; }

        public decimal CostPrice { get; set; }

        public int Stock { get; set; }

        public int MinimumQuantity { get; set; }

        public int MaximumQuantity { get; set; }

        public decimal Weight { get; set; }

        public string WeightUnit { get; set; }
    }
}
